asm_to_pil/src/lib.rs

```diff
@@ -206,21 +105,31 @@ impl<'a, T: FieldElement> ASMPILConverter<'a, T> {
             self.registers
                 .iter()
                 .filter_map(|(name, reg)| {
-                    reg.update_expression().map(|mut update| {
-                        if Some(name) == self.pc_name.as_ref() {
+                    reg.update_expression().map(|rhs| {
+                        let lhs = next_reference(name);
+                        use RegisterTy::*;
+                        let (lhs, rhs) = match reg.ty {
                             // Force pc to zero on first row.
-                            update = build_mul(
-                                build_sub(build_number(1u64), next_reference("first_step")),
-                                update,
-                            )
-                        }
+                            Pc => (
+                                lhs,
+                                build_mul(
+                                    build_sub(build_number(1u64), next_reference("first_step")),
```

Comment:
so `first_step` also shows up here, completely disconnected from the internal machine to compiler which is in a different crate. I think we need to address this now somehow before we get too far

Reply:
Agreed. I used other names in other compiler steps. I kept this name as this is what it was called before this PR. I could also rename to `_asm_to_pil_first_step`.